### PR TITLE
Network: fix descriptions & remove unused/duplicate definitions

### PR DESCRIPTION
**Pull request by anton-evseev**
* Remove duplicate definitions: `ProvisioningState`, `IpTag`
* Ensure all descriptions, definitions and enums start with capital letter to keep consistency across all network specs 
* Remove unused definition: `disableVpnEncryption` (this property is `boolean`, https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json#L2260)

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by sergey-shandar — APPROVED** [review]

**Comment by sergey-shandar** [comment]
@MikhailTryakhov could you review the PR?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/azureFirewall.json [hunk 1]
```diff
@@ -274,7 +274,7 @@
             },
             "provisioningState":{
                "description":"The provisioning state of the resource.",
-               "$ref":"#/definitions/ProvisioningState"
+               "$ref":"./network.json#/definitions/ProvisioningState"
```
Comment (MikhailTryakhov) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/azureFirewall.json]: Isn't it a breaking change on SDK level? we should not accept any breaking changes on already published API version

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/azureFirewall.json [hunk 2]
```diff
@@ -739,21 +739,6 @@
             "modelAsString":true
          }
       },
-      "ProvisioningState":{
```
Comment (MikhailTryakhov) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/azureFirewall.json]: Looks like breaking change. not acceptable for already published version. please revert and save for the next monthly branch

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json [hunk 3]
```diff
@@ -1605,7 +1605,7 @@
           ],
           "readOnly": true,
           "x-ms-enum": {
-            "name": "circuitConnectionStatus",
+            "name": "CircuitConnectionStatus",
```
Comment (MikhailTryakhov) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json]: Breaking?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/publicIpPrefix.json [hunk 4]
```diff
@@ -415,19 +415,6 @@
         },
         "description": "Response for ListPublicIpPrefixes API service call."
       },
-      "IpTag": {
```
Comment (MikhailTryakhov) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/publicIpPrefix.json]: Breaking :)

**Review by MikhailTryakhov — CHANGES_REQUESTED** [review]
No breaking changes need to be done for already published api version. please store changes need to be done for the next api version (next monthly branch)

**Review by KrisBash — CHANGES_REQUESTED** [review]
From an ARM perspective, these changes have no bearing on API schema/behavior and are not a concern. However, if the changes to definitions are SDK breaking, they should be saved for a new version.

**Comment by number213** [comment]
@MikhailTryakhov @KrisBash those changes are not breaking, you can check auto-PRs in SDK repos, only descriptions are changed in SDK code. Other fixes in this PR don't affect SDK code because:
- AutoRest adjusts casing of model names from lowercase to upper, but we still need to keep this casing consistent in Swagger.
- Duplicate definitions don't cause errors in AutoRest, one definition just rewrites the other. This could cause issues if someone edits one of them, making them different but having the same name.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/publicIpPrefix.json (hunk 4 above)
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/publicIpPrefix.json]: Nope

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json (hunk 3 above)
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json]: No

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/azureFirewall.json (hunk 2 above)
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/azureFirewall.json]: No, this is a duplicate

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/azureFirewall.json (hunk 1 above)
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/azureFirewall.json]: Removed duplicate definition and linked original. No change in SDK code

**Review by KrisBash — APPROVED** [review]

**Review by MikhailTryakhov — APPROVED** [review]

**Comment by number213** [comment]
@sergey-shandar everything has been resolved, could you please merge?

**Pull request merged**